jenkinsfile.groovy

```diff
@@ -0,0 +1,516 @@
+// using declarative pipeline: https://jenkins.io/doc/book/pipeline/syntax/
+@Library('PipelineUtils@feature-BL-8651-build-pipeline-fe-lambda')
+import dvsa.aws.mot.jenkins.pipeline.common.AWSFunctions
+import dvsa.aws.mot.jenkins.pipeline.common.RepoFunctions
+import dvsa.aws.mot.jenkins.pipeline.common.GlobalValues
+
+// global variables
+AWSFunctions awsFunctionsFactory = new AWSFunctions()
+RepoFunctions repoFunctionsFactory = new RepoFunctions()
+GlobalValues globalValuesFactory = new GlobalValues()
+Map<String, Map<String, String>> gitlab = globalValuesFactory.GITLAB_REPOS()
+Map<String, Map<String, String>> github = globalValuesFactory.GITHUB_REPOS()
+
+Map<String, String> params = [
+    branch              : APP_BRANCH,
+    tf_branch           : TF_BRANCH,
+    action              : ACTION,
+    environment         : ENVIRONMENT,
+    clean_workspace     : CLEAN_WORKSPACE,
+    selenium_hub_address: SELENIUM_HUB_ADDRESS,
+    skip_tests          : SKIP_TESTS
+]
+
+Map<String, Boolean> isNewVersion = [:]
+
+String deploymentJobName = "CVR_Deployment"
+String deploymentBucketJobName = "${deploymentJobName}_bucket"
+String jenkinsCtrlNodeLabel = "ctrl"
+String jenkinsBuildLabel = "cvr"
+String account = "dvsarecallsdev"
+String group = "dvsarecallsdev"
+String buildUser = ""
+String project = "cvr"
+String bucketPrefix = "terraformscaffold"
+String s3DeploymentBucket = ""
+String s3AssetsBucket = ""
+String frontendAppName = "frontend"
+String seleniumScreenshotsDir = "selenium-screenshots"
+String recallsApiGwUrl = ""
+String manifestVersion = ""
+String assetsBasePath = ""
+net.sf.json.JSON manifestContent
+
+
+def failure(String reason) {
+  currentBuild.result = "FAILURE"
+  log.fatal(reason)
+  error(reason)
+}
+
+String getCurrentCommit(String repoDir, String moduleDir = '.') {
+  String commit = ''
+  dir("${env.WORKSPACE}/${repoDir}") {
+    commit = sh (
+        script: "git log -n 1 --format=%H -- ${moduleDir}",
+        returnStdout: true
```

Comment:
would `git rev-parse HEAD` be simpler?

This would be a simple addition to the utils repo that we could then use elsewhere